lms/djangoapps/certificates/signals.py

```diff
@@ -0,0 +1,26 @@
+""" Signal handler for enabling self-generated certificates by default
+for self-paced courses.
+"""
+from celery.task import task
+from django.dispatch.dispatcher import receiver
+
+from certificates.models import CertificateGenerationCourseSetting
+from xmodule.modulestore.django import SignalHandler, modulestore
+
+
+@receiver(SignalHandler.course_published)
+def _listen_for_course_publish(sender, course_key, **kwargs):  # pylint: disable=unused-argument
+    """ Catches the signal that a course has been published in Studio and
+    enable the self-generated certificates by default for self-paced
+    courses.
+    """
+    enable_self_generated_certs.delay(course_key)
+
+
+@task()
+def enable_self_generated_certs(course_key):
+    """Enable the self-generated certificates by default for self-paced courses."""
+    course = modulestore().get_course(course_key)
```

Comment:
@waheedahmed , it's probably faster and more efficient to grab this information from the `CourseOverview` table